ipn/localapi/localapi.go

```diff
@@ -511,6 +513,35 @@ func (h *Handler) serveDERPMap(w http.ResponseWriter, r *http.Request) {
 	e.Encode(h.b.DERPMap())
 }
 
+// serveShortenExpiry sets the expiry date on the current machine to the
+// specified unix timestamp
+func (h *Handler) serveShortenExpiry(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "POST required", http.StatusMethodNotAllowed)
+		return
+	}
+
+	var expiryTime time.Time
+	if v := r.FormValue("expiry"); v != "" {
+		expiryInt, err := strconv.ParseInt(v, 10, 64)
+		if err != nil {
+			http.Error(w, "can't parse expiry time, expects a unix timestamp", http.StatusBadRequest)
+			return
+		}
+		expiryTime = time.Unix(expiryInt, 0)
+	} else {
+		http.Error(w, "missing 'expiry' parameter, a unix timestamp", http.StatusBadRequest)
+		return
+	}
+	err := h.b.ShortenExpiry(r.Context(), expiryTime)
+	if err != nil {
+		http.Error(w, err.Error(), 400)
```

Comment:
probably 500, not 400? Unknown actually.  also you used StatusBadRequest above but a number here.

Reply:
I set it as StatusBadRequest, the most likely scenario here is trying an expiry after the current expiry which isn't really 500-ish.